### DISCUSSION TITLE
Column grouping in current data-table

### DISCUSSION THREAD

**Discussion by sammykisina** [shadcn-ui/ui#1524] [Q&A] [answered]
Hello guys.
How can I have column grouping in the current data-table? If anyone has an idea please help.

**Comment by kaiiiiiiiii** [comment]
By just implementing it?

https://tanstack.com/table/v8/docs/examples/react/column-groups

**Reply by sammykisina** [reply]
I  had tried that sir before i asked and the columns did not match or go to their correct groups. I have also checked the utility classes on the styling, but the arrangement still looks wrong.
![image](https://github.com/shadcn-ui/ui/assets/47030001/e01fa474-6f0e-42a8-b520-85cefc81003d)

**Reply by anandagodini** [reply]
How did you got that more info at the top can send me the code

**Reply by haifan420** [reply]
I need it to bro, do you have the code?

**Comment by AmGhGitHub** [comment, marked as answer]
I have faced the same issue and I thought it is a bug in shadcn ui. Turned out to be a bug in my implementation. Make sure you have adjusted the table definition and include the colSpan={header.colSpan} in TableHead definition

```
          <TableHeader>
            {table.getHeaderGroups().map((headerGroup) => (
              <TableRow key={headerGroup.id} className="bg-zinc-300 text-black">
                {headerGroup.headers.map((header) => {
                  return (
                    // console.log(header),
                    <TableHead key={header.id} colSpan={header.colSpan}>
                      {header.isPlaceholder
                        ? null
                        : flexRender(
                            header.column.columnDef.header,
                            header.getContext()
                          )}
                    </TableHead>
                  );
                })}
              </TableRow>
            ))}
```

**Reply by sammykisina** [reply]
Thanks. Worked perfectly

**Reply by dev-myomyintaung** [reply]
Thanks. You saved my time for debugging.

**Reply by Gzbox** [reply]
Thanks

**Reply by rohithroshan-r** [reply]
Thank you @AmGhGitHub

**Reply by mwannags** [reply]
Thank you so much! Worked immediately!
